module_utils/network_lsr/argument_validator.py

```diff
@@ -271,7 +271,9 @@ def __init__(  # pylint: disable=too-many-arguments
     def _validate_impl(self, value, name):
         v = None
         try:
-            if isinstance(value, self.numeric_type):
```

Comment:
Please provide more context in the commit message. Currently, it needs more context to understand why it is mentioning 0 or 1.

Reply:
I explained as: 
```
In `./library/network_connections.py`, for ethtool coalesce setting,
there is unnecessary code snippet `if isinstance(setting, bool): setting = int(setting)`,
which convert the setting value from boolean into integer.
`int(False)` is evaluated to 0 , `int(True)` is evaluated to 1.
Therefore, we need to fix `ArgValidatorNum()` to reject boolean values
when integer values are expected for the setting. This bug fix
potentially breaks previously "working" playbooks (but realistically,
they were not working, because setting 0 or 1 was unlikely intended).
```

Reply:
This explanation does not seem right.

This is not about coalesce setting or what the mentioned code snippet does. It's primarily about how `ArgValidatorNum` should behave.

`ArgValidatorNum` was always supposed to give a value of type `self.numeric_type` -- except the default value (which might be `None` or even something else altogether).

There was a bug with 
```python
            if isinstance(value, self.numeric_type):
                 v = value
```
because `bool` is a subclass of `int` in python (which I was not aware of). But I don't thik that a `bool` should be accepted as a valid input for a number -- at least not by default (maybe in some cases). The bug was it would wrongly accept a boolean value, and it would wrongly pass it on (and not normalize it to `numeric_type`).


how about:
```python
            if self.numeric_type is not bool and isinstance(value, bool):
                # bool can (probably) be converted to self.numeric_type,
                # but here we don't want to accept a boolean value.
                pass
            elif isinstance(value, self.numeric_type):
                 v = self.numeric_type(value)
             else:
                 v2 = self.numeric_type(value)
                 if isinstance(value, Util.STRING_TYPE) or v2 == value:
                     v = v2
```

Reply:
Previously, in `ArgvalidaorNum` we did not normalize the boolean value into `numeric_type`. And those playbooks can still run without raising any error just because we did not wrongly pass boolean value when `numeric_type` is expected. Therefore, this bug fix did not break any playbooks at all.